exercises/csharp/TriviaGame/TriviaGame/Game.cs

```diff
@@ -2,190 +2,194 @@
-        LinkedList<string> scienceQuestions = new LinkedList<string>();
-        LinkedList<string> sportsQuestions = new LinkedList<string>();
-        LinkedList<string> rockQuestions = new LinkedList<string>();
+        readonly bool[] inPenaltyBox = new bool[6];
+              
+        readonly LinkedList<string> popQuestions = new LinkedList<string>();
+        readonly LinkedList<string> scienceQuestions = new LinkedList<string>();
+        readonly LinkedList<string> sportsQuestions = new LinkedList<string>();
+        readonly LinkedList<string> rockQuestions = new LinkedList<string>();
 
         int currentPlayer = 0;
         bool isGettingOutOfPenaltyBox;
 
         public Game()
         {
-            for (int i = 0; i < 50; i++)
+            for (int index = 0; index < 50; index++)
             {
-                popQuestions.AddLast("Pop Question " + i);
-                scienceQuestions.AddLast(("Science Question " + i));
-                sportsQuestions.AddLast(("Sports Question " + i));
-                rockQuestions.AddLast(CreateRockQuestion(i));
+                popQuestions.AddLast($"Pop Question {index}");
+                scienceQuestions.AddLast(($"Science Question {index}"));
+                sportsQuestions.AddLast(($"Sports Question {index}"));
+                rockQuestions.AddLast(CreateRockQuestion(index));
             }
         }
 
         public string CreateRockQuestion(int index)
         {
-            return "Rock Question " + index;
+            return $"Rock Question {index}";
         }
 
-        public bool IsPlayable()
-        {
-            return (HowManyPlayers() >= 2);
-        }
+        public bool IsPlayable() => HowManyPlayers() >= 2;
 
-        public bool Add(string playerName)
+        public bool AddPlayer(string name)
         {
-
-
-            players.Add(playerName);
+            players.Add(name);
             places[HowManyPlayers()] = 0;
-            purses[HowManyPlayers()] = 0;
+            coins[HowManyPlayers()] = 0;
             inPenaltyBox[HowManyPlayers()] = false;
 
-            Console.WriteLine(playerName + " was added");
-            Console.WriteLine("They are player number " + players.Count);
+            Console.WriteLine($"{name} was added");
+            Console.WriteLine($"They are player number {players.Count}");
             return true;
         }
 
-        public int HowManyPlayers()
+        public int HowManyPlayers() => players.Count;
+
+        public bool IsBackToStartOfBoard()
         {
-            return players.Count;
+            if (places[currentPlayer] > 11)
+            {
+                places[currentPlayer] = places[currentPlayer] - 12;
+            }
+            return true;
+        }
+        public void AdvanceOnGameBoard(int roll)
+        {
+            places[currentPlayer] = places[currentPlayer] + roll;
         }
 
-        public void Roll(int roll)
+        public void RollDice(int roll)
```

Comment:
I also called this RollDice, but someone pointed out that it does far more than just roll a dice